test/metabase/sync/field_values_test.clj

```diff
@@ -54,6 +54,27 @@
       (is (= [1 2 3 4]
              (venues-price-field-values))))))
 
+(deftest sync-should-properly-handle-last-used-at
+  (testing "Test that syncing will skip updating inactive FieldValues"
+    (db/update! FieldValues
+                (db/select-one-id FieldValues :field_id (mt/id :venues :price) :type :full)
+                :last_used_at (t/minus (t/offset-date-time) (t/days 20)))
+    (db/update! FieldValues (db/select-one-id FieldValues :field_id (mt/id :venues :price) :type :full) :values [1 2 3])
+    (is (= (repeat 2 {:errors 0, :created 0, :updated 0, :deleted 0})
+           (sync-database!' "update-field-values" (data/db))))
+    (is (= [1 2 3] (venues-price-field-values)))
+    (testing "Fetching field values causes an on-demand update and marks Field Values as active"
+      (is (partial= {:values [[1] [2] [3] [4]]}
+                    (mt/user-http-request :rasta :get 200 (format "field/%d/values" (mt/id :venues :price)))))
+      (let [last-used-at (db/select-one-field :last_used_at FieldValues :field_id (mt/id :venues :price) :type :full)]
+        (is (t/after? last-used-at (t/minus (t/offset-date-time) (t/hours 2))))
+        (testing "Fetching again updates last-used-at"
+          (Thread/sleep 500) ;; Let clock advance to make sure last-used-at is different
+          (mt/user-http-request :rasta :get 200 (format "field/%d/values" (mt/id :venues :price)))
+          (is (t/after?
```

Comment:
worth having a test for sync to pick it up after `last-used-at` is updated?

Reply:
Changed this to stop relying on the time comparison as it was too flaky, instead used the sync as you suggested.